keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/ArtifactListener.kt

```diff
@@ -53,17 +54,19 @@ class ArtifactListener(
       .forEach { artifact ->
         if (repository.isRegistered(artifact.name, artifact.artifactType)) {
           val artifactSupplier = artifactSuppliers.supporting(artifact.artifactType)
-          val version = artifactSupplier.getFullVersionString(artifact)
-          var status = artifactSupplier.getReleaseStatus(artifact)
-          log.info("Registering version {} ({}) of {} {}", version, status, artifact.name, artifact.type)
-          repository.storeArtifact(artifact.name, artifact.artifactType, version, status)
+          val enrichedArtifact = artifactSupplier.addMetadata(artifact.normalized())
+
+          log.info("Registering version {} (status={}) of {} artifact {}",
+            artifact.version, artifact.status, artifact.type, artifact.name)
+
+          repository.storeArtifactVersion(enrichedArtifact)
```

Comment:
This method now persists both the "basic" properties of the artifact and the metadata in one database write.